### PR TITLE
BetterC & Mir rework: @nogc SharedFreeList, BitmappedBlock and imports cleanup

### PR DESCRIPTION
**Pull request by 9il**
... to be continued in the next PR.

### CODE REVIEW

File: source/stdx/allocator/building_blocks/free_list.d [hunk 1]
```diff
@@ -782,21 +777,23 @@ struct SharedFreeList(ParentAllocator,
...
             // one shot.
             void setBounds(size_t low, size_t high) shared
             {
-                enforce(low <= high && high >= (void*).sizeof);
-                enforce(cas(&_min, chooseAtRuntime, low),
-                    "SharedFreeList.min must be initialized exactly once.");
-                enforce(cas(&_max, chooseAtRuntime, high),
-                    "SharedFreeList.max must be initialized exactly once.");
+                if (!(low <= high && high >= (void*).sizeof))
+                    throw excBounds;
```
Comment (wilzbach) [inline source/stdx/allocator/building_blocks/free_list.d:792]: Why not add an enforce template that will create the static exception?

File: source/stdx/allocator/internal.d [hunk 2]
```diff
@@ -1259,7 +1259,7 @@ pure nothrow @safe /* @nogc */ unittest
     import core.stdc.stdlib : malloc;
     void[] buf = malloc(Node.sizeof)[0 .. Node.sizeof];
 
-    import std.conv : emplace;
+    import stdx.allocator.internal : emplace;
```
Comment (wilzbach) [inline source/stdx/allocator/internal.d:1262]: I guess this import could be removed.

**Review by wilzbach — APPROVED** [review]
Would prefer if there was a little helper for exceptions, s.t. they are less boilerplate.

File: source/stdx/allocator/building_blocks/free_list.d (hunk 1 above)
Comment (9il) [reply source/stdx/allocator/building_blocks/free_list.d:792]: Maybe will add it in the future. Mir Exception is declared in mir-runtime, need to play with DIP1008 more.

Comment (9il) [reply source/stdx/allocator/building_blocks/free_list.d:792]: I have used such enforce in the production. How would you name it?

Comment (wilzbach) [reply source/stdx/allocator/building_blocks/free_list.d:792]: I would actually use `enforce`, but probably e.g. `massert` would work too.
What's more interesting is how such a template would be. A few ideas of my head:

a) `enforce!"SharedFreeList.max must .."(!(low <= high && high >= (void*).sizeof))` (pass exception string as template paramter)
b) `enforce!"SharedFreeList"(!(low <= high && high >= (void*).sizeof), "SharedFreeList.max must ..")` (pass unique id as template parameter + custom exception string)
c) `enforce(!(low <= high && high >= (void*).sizeof), "SharedFreeList.max must ..")` (uses function caller name as automatic template argument)

**Pull request merged**